components/lib/auth/src/main/java/mozilla/components/lib/auth/BiometricUtils.kt

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.lib.auth
+
+import android.content.Context
+import android.os.Build
+import androidx.biometric.BiometricManager
+
+/**
+ * Utility class for BiometricPromptAuth
+ */
+
+fun Context.canUseBiometricFeature(): Boolean {
+    return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+        val manager = BiometricManager.from(this)
+        return BiometricUtils.canUseFeature(manager)
+    } else {
+        false
+    }
+}
+
+internal object BiometricUtils {
+
+    /**
+     * Checks if the appropriate SDK version and hardware capabilities are met to use the feature.
+     */
+    internal fun canUseFeature(manager: BiometricManager): Boolean {
+        return isHardwareAvailable(manager) && isEnrolled(manager)
+    }
+
+    /**
+     * Checks if the hardware requirements are met for using the [BiometricManager].
+     */
+    private fun isHardwareAvailable(biometricManager: BiometricManager): Boolean {
```

Comment:
Sorry, maybe I wasn't clear: you can still write tests for these easily I think, but you don't have to make the utils class public.

So the methods can be internal:

```kotlin
    internal fun isHardwareAvailable(biometricManager: BiometricManager): Boolean { }
    internal fun isEnrolled(biometricManager: BiometricManager): Boolean { }
```

And the tests can take a mocked biometric manager to write the tests for those methods:
```kotlin

    @Test
    fun `isHardwareAvailable is true based on AuthenticationStatus`() {
        val manager: BiometricManager = mock {
            whenever(canAuthenticate(BiometricManager.Authenticators.BIOMETRIC_WEAK))
                .thenReturn(BiometricManager.BIOMETRIC_SUCCESS)
                .thenReturn(BiometricManager.BIOMETRIC_ERROR_HW_UNAVAILABLE)
                .thenReturn(BiometricManager.BIOMETRIC_ERROR_NO_HARDWARE)
        }

        assertTrue(BiometricUtils.isHardwareAvailable(manager))
        assertFalse(BiometricUtils.isHardwareAvailable(manager))
        assertFalse(BiometricUtils.isHardwareAvailable(manager))
    }

    @Test
    fun `isEnrolled is true based on AuthenticationStatus`() {
        val manager: BiometricManager = mock {
            whenever(canAuthenticate(BiometricManager.Authenticators.BIOMETRIC_WEAK))
                .thenReturn(BiometricManager.BIOMETRIC_SUCCESS)
        }
        assertTrue(BiometricUtils.isEnrolled(manager))
    }
```

The test you added for `canUseFeature` is good enough coverage for that method.

Reply:
Done